providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/GoFeatureFlagProvider.java

```diff
@@ -173,60 +108,77 @@ public ProviderEvaluation<Value> getObjectEvaluation(
+
+        if (options.getEnableCache() == null || options.getEnableCache()) {
+            this.cacheCtrl = CacheController.builder().options(options).build();
+
+            if (!this.options.isDisableDataCollection()) {
+                this.dataCollectorHook = new DataCollectorHook(DataCollectorHookOptions.builder()
+                        .flushIntervalMs(options.getFlushIntervalMs())
+                        .gofeatureflagController(this.gofeatureflagController)
+                        .maxPendingEvents(options.getMaxPendingEvents())
+                        .build());
+                this.hooks.add(this.dataCollectorHook);
+            }
+            this.flagChangeDisposable =
+                    this.startCheckFlagConfigurationChangesDaemon();
         }
         state = ProviderState.READY;
+        super.emitProviderReady(ProviderEventDetails.builder().message("Provider is ready to call the API").build());
         log.info("finishing initializing provider, state: {}", state);
     }
 
+
+    /**
+     * startCheckFlagConfigurationChangesDaemon is a daemon that will check if the flag configuration has changed.
+     *
+     * @return Disposable - the subscription to the observable
+     */
+    @NotNull
+    private Disposable startCheckFlagConfigurationChangesDaemon() {
+        long pollingIntervalMs = options.getFlagChangePollingIntervalMs() != null
+                ? options.getFlagChangePollingIntervalMs() : DEFAULT_POLLING_CONFIG_FLAG_CHANGE_INTERVAL_MS;
+
+        PublishSubject<Object> stopSignal = PublishSubject.create();
+        Observable<Long> intervalObservable = Observable.interval(pollingIntervalMs, TimeUnit.MILLISECONDS);
+        Observable<ConfigurationChange> apiCallObservable = intervalObservable
+                // as soon something is published in stopSignal, the interval will stop
+                .takeUntil(stopSignal)
+                .flatMap(tick -> Observable.fromCallable(() -> this.gofeatureflagController.configurationHasChanged())
+                        .onErrorResumeNext(e -> {
+                            log.error("error while calling flag change API", e);
+                            if (e instanceof ConfigurationChangeEndpointNotFound) {
+                                // emit an item to stop the interval to stop the loop
+                                stopSignal.onNext(new Object());
+                            }
+                            return Observable.empty();
+                        }))
+                .subscribeOn(Schedulers.io());
+
+        Disposable disposable = apiCallObservable
+                .subscribe(
+                        response -> {
+                            if (response == ConfigurationChange.FLAG_CONFIGURATION_UPDATED) {
+                                log.info("clean up the cache because the flag configuration has changed");
+                                this.cacheCtrl.invalidateAll();
```

Comment:
the response does not contain the flag that changed, to invalidate only them ?

Reply:
Not on this version, the API returns only a hash.